File: cloudtimes-business/src/main/java/com/cloudtimes/app/controller/agent/CtAgentDividendSettlementController.java
package com.cloudtimes.app.controller.agent;

import com.cloudtimes.account.domain.CtAgentDividendSettlement;
import com.cloudtimes.account.service.ICtAgentDividendSettlementService;
import com.cloudtimes.common.annotation.Log;
import com.cloudtimes.common.core.controller.BaseController;
import com.cloudtimes.common.core.domain.AjaxResult;
import com.cloudtimes.common.core.page.TableDataInfo;
import com.cloudtimes.common.enums.BusinessType;
import io.swagger.annotations.Api;
import io.swagger.annotations.ApiOperation;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

import java.util.List;

/**
 * 分润结算审核Controller
 * 
 * @author 沈兵
 * @date 2023-02-03
 */
@RestController
@RequestMapping("/agent/dividend_settlement")
@Api(tags = "代理-分润结算")
public class CtAgentDividendSettlementController extends BaseController
{
    @Autowired
    private ICtAgentDividendSettlementService ctAgentDividendSettlementService;

    /**
     * 查询分润结算审核列表
     */
    @GetMapping("/list")
    @ApiOperation("查询分润结算审核列表")
    public TableDataInfo list(CtAgentDividendSettlement ctAgentDividendSettlement)
    {
        startPage();
        List<CtAgentDividendSettlement> list = ctAgentDividendSettlementService.selectCtAgentDividendSettlementList(ctAgentDividendSettlement);
        return getDataTable(list);
    }

    /**
     * 获取分润结算审核详细信息
     */
    @GetMapping(value = "/{id}")
    @ApiOperation("获取分润结算审核详细信息")
    public AjaxResult getInfo(@PathVariable("id") String id)
    {
        return AjaxResult.success(ctAgentDividendSettlementService.selectCtAgentDividendSettlementById(id));
    }

    /**
     * 新增分润结算审核
     */
    @Log(title = "分润结算审核", businessType = BusinessType.INSERT)
    @PostMapping
    @ApiOperation("新增分润结算")
    public AjaxResult add(@RequestBody CtAgentDividendSettlement ctAgentDividendSettlement)
    {
        return toAjax(ctAgentDividendSettlementService.insertCtAgentDividendSettlement(ctAgentDividendSettlement));
    }

    /**
     * 修改分润结算审核
     */
    @Log(title = "分润结算审核", businessType = BusinessType.UPDATE)
    @PutMapping
    @ApiOperation("修改分润结算")
    public AjaxResult edit(@RequestBody CtAgentDividendSettlement ctAgentDividendSettlement)
    {
        return toAjax(ctAgentDividendSettlementService.updateCtAgentDividendSettlement(ctAgentDividendSettlement));
    }
}

File: cloudtimes-business/src/main/java/com/cloudtimes/app/controller/agent/CtAgentDividendSettlementController.kt
package com.cloudtimes.app.controller.agent

import com.cloudtimes.agent.domain.CtAgentDividendSettlement
import com.cloudtimes.agent.dto.request.AgentDividendRequest
import com.cloudtimes.agent.dto.request.StoreDividendRequest
import com.cloudtimes.agent.service.ICtAgentDividendSettlementService
import com.cloudtimes.common.core.controller.BaseController
import com.cloudtimes.common.core.domain.AjaxResult
import com.cloudtimes.common.core.domain.RestPageResult
import com.cloudtimes.common.core.domain.RestResult
import io.swagger.annotations.Api
import io.swagger.annotations.ApiOperation
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import javax.validation.Valid

// 泛型具体化
class DividendSettlementPage: RestPageResult<CtAgentDividendSettlement>()
class DividendDetail: RestResult<CtAgentDividendSettlement>()

/**
 * 分润结算审核Controller
 *
 * @author 沈兵
 * @date 2023-02-03
 */
@RestController
@RequestMapping("/agent/dividend_settlement")
@Api(tags = ["代理-分润结算"])
class CtAgentDividendSettlementController : BaseController() {
    @Autowired
    private lateinit var dividendSettlementService: ICtAgentDividendSettlementService

    @PostMapping("/list_store_dividend")
    @ApiOperation("查询店铺分润列表")
    fun listStoreDividend(@Valid @RequestBody request: StoreDividendRequest): DividendSettlementPage {
        startPage(request.pageNum, request.pageSize)
        val list = dividendSettlementService.selectStoreDividendList(request)
        val page = getDataTable(list)
        return DividendSettlementPage().apply {
            data = list
            total = page.total
        }
    }

    @PostMapping("/detail")
    @ApiOperation("查询店铺分润详情")
    fun detail(@Valid @RequestBody request: AgentDividendRequest): DividendDetail {
        val detail = dividendSettlementService.agentDividendDetail(request)
        return DividendDetail().apply {
            data = detail
        }
    }

    @PostMapping("/agent_approve")
    @ApiOperation("店铺分润代理确认")
    fun agentApprove(@Valid @RequestBody request: AgentDividendRequest): AjaxResult {
        dividendSettlementService.agentApproveDividend(request)
        return AjaxResult.success()
    }
}